# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserBridgeParent.cpp
# Commit: e1fd1fa67a65
# Full Hash: e1fd1fa67a6554ae8dc206a6fed9e5405a0bf778
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-04-01 03:12:48
# Regressor Bug: 1699721
# File Overlap Count: 1
# Description:
#   Bug 1699721 - Part 2: Track BrowserParent lifecycles during process switches, r=kmag
#   
#   This patch contains a large number of changes around the process switching
#   mechanism in order to avoid issues which are caused by a mismatched
#   understanding of the state of the process switch between processes in the
# ==============================================================================

diff -r 2b3a6042d48e -r e1fd1fa67a65 dom/ipc/BrowserBridgeParent.cpp
--- a/dom/ipc/BrowserBridgeParent.cpp	Wed Mar 31 16:51:58 2021 +0000
+++ b/dom/ipc/BrowserBridgeParent.cpp	Wed Mar 31 16:51:58 2021 +0000
@@ -41,6 +41,10 @@
     return NS_ERROR_UNEXPECTED;
   }
 
+  MOZ_DIAGNOSTIC_ASSERT(
+      !browsingContext->GetBrowserParent(),
+      "BrowsingContext must have had previous BrowserParent cleared");
+
   // Unfortunately, due to the current racy destruction of BrowsingContext
   // instances when Fission is enabled, while `browsingContext` may not be
   // discarded, an ancestor might be.
@@ -108,6 +112,8 @@
   mBrowserParent->SetOwnerElement(aParentBrowser->GetOwnerElement());
   mBrowserParent->InitRendering();
 
+  GetBrowsingContext()->SetCurrentBrowserParent(mBrowserParent);
+
   windowParent->Init();
   return NS_OK;
 }